microcosm/config/model.py

```diff
@@ -98,26 +98,44 @@ def __init__(
         self.default_factory = default_factory
         self.mock_value = mock_value
         self.required = required
-        self.nullable = nullable
-
-        # Warn when the user doesn't provide exactly one of
-        #
-        # [`required=True`, `default_value=...`]
-        #
-        # When the user is not using `default_factory`, this could be legacy
-        # code, so we just warn. In the future this will be an error, but for
-        # now we just warn in those situations which could occur as of the time
-        # this was introduced
+        self.nullable = nullable or (default_value is None)
+
+        if kwargs:
+            warn(
+                f"Unsupported `Requirement` args {kwargs}.  "
+                "Support for arbitrary arguments to `Requirement` will be dropped",
+                category=FutureWarning,
+            )
+
         if default_factory is None and required == (default_value is not UNSET):
+            # Warn when the user doesn't provide exactly one of
+            #
+            # [`required=True`, `default_value=...`]
+            #
+            # When the user is not using `default_factory`, this could be legacy
+            # code, so we just warn. In the future we will remove this clause
+            # and just use the error clause below, but for now we just warn in
+            # those situations which could occur as of the time this was
+            # introduced
             warn(
                 "Must either specify `required=True` or provide default value.",
                 category=FutureWarning,
             )
         elif any([
+            # Must either require a value or provide a default
+            all([
+                not required,
+                default_value is UNSET,
+                default_factory is None,
+            ]),
```

Comment:
Forgot this case.  Wasn't failing tests because it was caught by warning above, but will need it when we switch to always erroring